src/ophys_etl/modules/suite2p_registration/suite2p_utils.py

```diff
@@ -0,0 +1,116 @@
+import h5py
+import numpy as np
+from suite2p.registration.register import pick_initial_reference
+from suite2p.registration.rigid import (apply_masks, compute_masks, phasecorr,
+                                        phasecorr_reference, shift_frame)
+
+
+def load_initial_frames(file_path: str,
+                        h5py_key: str,
+                        nimg_init: int) -> np.ndarray:
+    """Load a subset of frames from the hdf5 data specified by file_path.
+
+    Parameters
+    ----------
+    file_path : str
+        Location of the raw 2Photon, HDF5 data to load.
+    h5py_key : str
+        Name of the dataset to load from the HDF5 file.
+    nimg_init : int
+        Number of frames to load from the input HDF5 data.
+
+    Returns
+    -------
+    frames : array-like, (nimg_init, Ly, Lx)
+        Selected frames from the input raw data linearly spaced in index of the
+        time axis. If nimg_init > nframes, a number of frames equal to nframes
+        is returned.
+    """
+    with h5py.File(file_path, "r") as hdf5_file:
+        # Get a set of linear spaced frames to load from disk.
+        tot_frames = hdf5_file[h5py_key].shape[0]
+        requested_frames = np.linspace(0,
+                                       tot_frames,
+                                       1 + min(nimg_init, tot_frames),
+                                       dtype=int)[:-1]
+        frames = hdf5_file[h5py_key][requested_frames]
+    return frames
+
+
+def compute_reference(frames: np.ndarray,
+                      niter: int,
+                      maxregshift: float,
+                      smooth_sigma: float,
+                      smooth_sigma_time: float) -> np.ndarray:
+    """Computes a stacked reference image from the input frames.
+
+    Modified version of Suite2P's compute_reference function with no updating
+    of input frames. Picks initial reference then iteratively aligns frames to
+    create reference. Does not reproduce the Suite2p 1Photon code path.
```

Comment:
What does "1photon code path" mean? I guess it would be clear if I looked at the Suite2p codebase but I don't think readers of this codebase would be familiar with it.

Reply:
How about "This code does not reproduce the processing suite2p does data from 1Photon scopes. As such, if processing 1Photon, the user should use the suite2p reference image creation."?